sktime/forecasting/regularized_var.py

```diff
@@ -0,0 +1,224 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements Regularized (L1/L2) VAR using cvxpy."""
+
+# __author__ = [meraldoantonio]
+
+import numpy as np
+import pandas as pd
+
+from sktime.forecasting.base import BaseForecaster
+
+
+class RegularizedVAR(BaseForecaster):
+    """Custom forecaster using VAR with regularization.
+
+    Parameters
+    ----------
+    lags : int
+        The number of lagged values to include in the model.
+    L1_penalty : float, optional (default=0.0)
+        The L1 regularization penalty.
+    L2_penalty : float, optional (default=0.0)
+        The L2 regularization penalty.
+    """
```

Comment:
do you have a sci reference that recommends fitting VAR like this?

Reply:
I was looking at the following papers that regularize VAR using L1/L2:
https://arxiv.org/abs/2105.00860
https://strathprints.strath.ac.uk/57829/1/Cavalcante_etal_WE2016_LASSO_vector_autoregression_structures_for_very_short_term_wind_power_forecasting.pdf

Reply:
thanks! Just to avoid me sifting through pages and pages of dense math if you have already done that, can you point to the place where this specific algorithm (tabularization and then minimization) is suggested?

Reply:
Actually, the papers I found didn't explicitly suggest the specific algorithm of tabularization followed by minimization. I couldn't find any papers that did. The two papers I reviewed simply described the use of L1/L2 regularization in LASSO and their advantages without going through tabularization. However, the algorithm we used is mathematically equivalent to a "direct optimization, such as the previous version I implemented using CVXPY.

For instance, Section 2 of [this paper](https://strathprints.strath.ac.uk/57829/1/Cavalcante_etal_WE2016_LASSO_vector_autoregression_structures_for_very_short_term_wind_power_forecasting.pdf) describes the motivation for using LASSO VAR:

---
The VAR model allows a simultaneous forecast of the wind generation in several neighboring sites combining time
series information. However, forecasting with VAR models may be intractable for high dimensional data since the nonsparse coefficients matrix grows quadratically with the number of series included in the model. In order to overcome this limitation, in [28] it is proposed the combination of LASSO and VAR frameworks, which is further explored in this paper for very short-term forecasting of wind power.
---

Similarly, Section 2 of [this paper](https://arxiv.org/pdf/2105.00860) provides a discussion of the ridge penalty and the ridge VAR estimator.